packages/react-components/react-message-bar/library/src/components/MessageBarGroup/renderMessageBarGroup.tsx

```diff
@@ -14,18 +14,17 @@ export const renderMessageBarGroup_unstable = (state: MessageBarGroupState) => {
 
   return (
     <state.root>
-      <TransitionGroup component={null}>
-        {state.children.map(child => (
-          <MessageBarTransition
-            animate={state.animate}
-            key={child.key}
-            enterClassName={state.enterStyles}
-            exitClassName={state.exitStyles}
-          >
-            {child}
-          </MessageBarTransition>
-        ))}
-      </TransitionGroup>
+      <PresenceGroup>
+        {state.children.map(child =>
+          state.animate === 'both' ? (
+            // enter with slide and fade; exit with fade
+            <SlideInFadeOut key={child.key}>{child}</SlideInFadeOut>
+          ) : (
+            // no enter motion; exit with fade
+            <FadeOut key={child.key}>{child}</FadeOut>
+          ),
+        )}
```

Comment:
IMO, the motion should be inside `MessageBar` as it's where it belongs.
This will also avoid custom child iteration & make the component more flexible.

Reply:
@layershifter Perhaps that would be better, but I'm trying to make this migration with a minimal footprint. I'm following the existing structure where the transition code is inside `MessageBarGroup`, and the `state.animate` flag is in `MessageBarGroupState`. Also, the only Storybook examples that show the animations are via `MessageBarGroup`. So I'm not at the point where I'm trying to rethink what it would mean for a `MessageBar` to animate itself, outside of a `MessageBarGroup`.

Reply:
> Also, the only Storybook examples that show the animations are via MessageBarGroup. So I would rather not, in this PR, start rethinking what it would mean for a MessageBar to animate itself, outside of a MessageBarGroup.

That's actually makes sense 👍 

----

There is one unobvious issue: when `state.animate` changes  ➡️ an `elementType` will change and React will remount a part of a tree ➡️ children will loose their state 💥

As we have already custom components, can we create a compound component and pass a param to it? I.e.:

```
        {state.children.map(child => (
            <MessageBarMotion
              animate={state.animate}
              key={child.key}
            >
                {child}
            </MessageBarMotion>
          )
        )}
``` 

Then we will avoid the remount problem.

Reply:
Creating that compound component seems easy enough. 

I will say that in practice, changing `state.animate` would very rarely change; it would almost always be configured on startup and left that way. The only use case I can think of for changing it on the fly is for testing the property itself.

Reply:
@layershifter Done in https://github.com/microsoft/fluentui/pull/33465/commits/834480dc679ed3f5e2a786ae50ca294096a51888

Reply:
@robertpenner sorry, it seems I was not clear. We need to do the following:

```tsx
const MessageBarMotion = createPresence<{ animate: /* --- */ }>({ animate }) => {
  if (animate === /* --- *) {
    return atomA()
  }

  return atomB()
}
```

So an element in React tree will not change. There is more on this topic: https://legacy.reactjs.org/docs/reconciliation.html#elements-of-different-types

Reply:
@layershifter I see what you mean. Addressed in https://github.com/microsoft/fluentui/pull/33465/commits/8fdaa1d60e024434e6dc6e8a0b75c928d9e4241d